### PR TITLE
batcheval: keep correct stats in AddSSTable without overlap

### PR DESCRIPTION
**Pull request by tbg**
For distributed index backfill, we recently switched to allowing the stats
of the data in the SST to be blindly added to that of the range
(thus introducing ContainsEstimates=true). However, if the keyspace being
ingested into is empty (such as usually the case during IMPORT/RESTORE), it
seems a little silly to make all the consistency checks for the next 24h
twice as expensive. Probe whether the keyspace is empty and if so, don't
pretend to introduce an approximation - the stats in this case are exact.

Release note: None

### CODE REVIEW

**Review by dt — APPROVED** [review]
Though should note that the assertion that IMPORT SSTs don't overlap is only true of distsql sorted import -- direct import will probably be the default in 19.2 and it will likely make overlapping SSTs.

File: pkg/storage/replica_raft.go [hunk 1]
```diff
@@ -1860,8 +1860,16 @@ func (r *Replica) processRaftCommand(
 		// succeeded. Note that we have taken precautions during command
 		// evaluation to avoid having mutations in the WriteBatch that affect
 		// the SSTable. Not doing so could result in order reversal (and missing
-		// values) here. If the key range we are ingesting into isn't empty,
-		// we're not using AddSSTable but a plain WriteBatch.
+		// values) here.
+		//
+		// Note that in the event of an ill-timed crash, the data may have been
+		// ingested (i.e. become "visible") without the Raft command having
+		// marked as applied. However, the node will only serve reads once a new
```
Comment (dt) [inline pkg/storage/replica_raft.go]: Is this true with follower reads?

Comment (tbg) [reply pkg/storage/replica_raft.go]: Follower reads will see extra data, which is weird (and the data could be out-of-order with the raft log...) anyway, hard to figure what exactly a problem here would be, but closed timestamps would definitely go down the drain and everything is pretty undefined with sst ingestion anyway. I rearranged the comment and clarified.

**Comment by tbg** [comment]
Superseded by many changes in cmd_add_sstable.go that have landed since.

**Pull request closed without merging**